File: app/src/main/java/com/originb/inkwisenote/modules/smartnotes/data/SmartBooksDao.java
package com.originb.inkwisenote.modules.smartnotes.data;

import androidx.room.*;

import java.util.List;
import java.util.Set;

@Dao
public interface SmartBooksDao {
    @Query("SELECT * FROM smart_books WHERE book_id = :bookId")
    SmartBookEntity getSmartbooksWithMatchingTitle(long bookId);

    @Query("SELECT * FROM smart_books WHERE title like :title")
    List<SmartBookEntity> getSmartbooksWithMatchingTitle(String title);

    @Query("SELECT * FROM smart_books ")
    List<SmartBookEntity> getAllSmartBooks();

    @Query("SELECT * FROM smart_books WHERE book_id IN (:bookIds)")
    List<SmartBookEntity> getSmartBooks(Set<Long> bookIds);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertSmartBook(SmartBookEntity smartBookEntity);

    @Update
    int updateSmartBook(SmartBookEntity smartBookEntity);

    @Query("DELETE FROM smart_books WHERE book_id = :bookId")
    void deleteSmartBook(long bookId);
}

File: app/src/main/java/com/originb/inkwisenote2/modules/smartnotes/data/SmartBooksDao.kt
package com.originb.inkwisenote2.modules.smartnotes.data

import androidx.room.*

@Dao
interface SmartBooksDao {
    @Query("SELECT * FROM smart_books WHERE book_id = :bookId")
    fun getSmartbooksWithMatchingTitle(bookId: Long): SmartBookEntity?

    @Query("SELECT * FROM smart_books WHERE title like :title")
    fun getSmartbooksWithMatchingTitle(title: String?): List<SmartBookEntity?>?

    @get:Query("SELECT * FROM smart_books ")
    val allSmartBooks: List<SmartBookEntity?>?

    @Query("SELECT * FROM smart_books WHERE book_id IN (:bookIds)")
    fun getSmartBooks(bookIds: Set<Long?>?): List<SmartBookEntity?>?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertSmartBook(smartBookEntity: SmartBookEntity?): Long

    @Update
    fun updateSmartBook(smartBookEntity: SmartBookEntity?): Int

    @Query("DELETE FROM smart_books WHERE book_id = :bookId")
    fun deleteSmartBook(bookId: Long)
}